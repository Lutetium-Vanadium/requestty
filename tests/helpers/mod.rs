use std::io;

use ui::{backend, style};

pub struct SnapshotOnFlushBackend {
    backend: backend::TestBackend,
}

impl SnapshotOnFlushBackend {
    #[allow(dead_code)]
    pub fn new(size: backend::Size) -> Self {
        Self {
            backend: backend::TestBackend::new(size),
        }
    }

    #[allow(dead_code)]
    pub fn new_with_layout(size: backend::Size, layout: ui::layout::Layout) -> Self {
        Self {
            backend: backend::TestBackend::new_with_layout(size, layout),
        }
    }
}

impl io::Write for SnapshotOnFlushBackend {
    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
        self.backend.write(buf)
    }

    fn flush(&mut self) -> io::Result<()> {
        ui::assert_backend_snapshot!(self.backend);
        Ok(())
    }
}

impl backend::DisplayBackend for SnapshotOnFlushBackend {
    fn set_attributes(&mut self, attributes: style::Attributes) -> io::Result<()> {
        self.backend.set_attributes(attributes)
    }

    fn set_fg(&mut self, color: style::Color) -> io::Result<()> {
        self.backend.set_fg(color)
    }

    fn set_bg(&mut self, color: style::Color) -> io::Result<()> {
        self.backend.set_bg(color)
    }
}

impl backend::Backend for SnapshotOnFlushBackend {
    fn enable_raw_mode(&mut self) -> io::Result<()> {
        self.backend.enable_raw_mode()
    }

    fn disable_raw_mode(&mut self) -> io::Result<()> {
        self.backend.disable_raw_mode()
    }

    fn hide_cursor(&mut self) -> io::Result<()> {
        self.backend.hide_cursor()
    }

    fn show_cursor(&mut self) -> io::Result<()> {
        self.backend.show_cursor()
    }

    fn get_cursor_pos(&mut self) -> io::Result<(u16, u16)> {
        self.backend.get_cursor_pos()
    }

    fn move_cursor_to(&mut self, x: u16, y: u16) -> io::Result<()> {
        self.backend.move_cursor_to(x, y)
    }

    fn move_cursor(&mut self, direction: backend::MoveDirection) -> io::Result<()> {
        self.backend.move_cursor(direction)
    }

    fn scroll(&mut self, dist: i16) -> io::Result<()> {
        self.backend.scroll(dist)
    }

    fn clear(&mut self, clear_type: backend::ClearType) -> io::Result<()> {
        self.backend.clear(clear_type)
    }

    fn size(&self) -> io::Result<backend::Size> {
        self.backend.size()
    }
}
