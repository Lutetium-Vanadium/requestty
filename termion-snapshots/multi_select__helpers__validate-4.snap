---
source: tests/helpers/mod.rs
expression: self.backend

---
┌──────────────────────────────────────────────────┐
│[38;5;10m? [1m[39mmulti select[22m [38;5;8m(Press <space> to select, <a> to to[39m│
│[38;5;8mggle all, <i> to invert selection)[39m                │
│  [38;5;8m  [39mChoice 0                                      │
│  [38;5;8mSeparator 1[39m                                     │
│  [38;5;8m──────────────[39m                                  │
│[38;5;6m❯ [38;5;10m✔ [38;5;6mChoice 3[39m                                      │
│  [38;5;8m  [39mChoice 4                                      │
│  [38;5;8mSeparator 5[39m                                     │
│  [38;5;8m  [39mChoice 6                                      │
│  [38;5;8m  [39mChoice 7                                      │
│  [38;5;8m  [39mChoice 8                                      │
│  [38;5;8m  [39mChoice 9                                      │
│[38;5;1m✖[39m At least 2 items must be checked. 1 items were  │
│checked                                           │
│                                                  │
│                                                  │
│                                                  │
│                                                  │
│                                                  │
│                                                  │
└──────────────────────────────────────────────────┘
